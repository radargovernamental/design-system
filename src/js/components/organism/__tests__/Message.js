import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Message from '../Message';

configure({ adapter: new Adapter() });

describe('Message', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Message>Lorem</Message>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with children inside it', () => {
    const wrapper = shallow(<Message><div>Lorem <strong>Ipsum</strong></div></Message>);
    expect(wrapper.children().html()).toBe('<span><div>Lorem <strong>Ipsum</strong></div></span>');
  });
});
