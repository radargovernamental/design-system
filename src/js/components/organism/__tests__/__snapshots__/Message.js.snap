// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    className=""
    close={null}
    danger={false}
    gray={false}
    icon={null}
    primary={false}
    success={false}
    warning={false}
>
    Lorem
</Message>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <span
          className={undefined}
>
          Lorem
</span>,
        null,
      ],
      "className": " ",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": undefined,
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "span",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <span
            className={undefined}
>
            Lorem
</span>,
          null,
        ],
        "className": " ",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "span",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
